core/src/main/java/org/mule/routing/MessageFilter.java

```diff
@@ -188,4 +190,13 @@ public void dispose()
 
         LifecycleUtils.disposeIfNeeded(filter, LOGGER);
     }
+
+    @Override
+    public void addMessageProcessorPathElements(MessageProcessorPathElement pathElement)
+    {
+        if (!(next instanceof MessageProcessorChain))
```

Comment:
In the second test case (the _for each_ one) the DefaultMessageProcessorChainBuilder sets the test component as the next operator of the MessageFilter, but as a DefaultMessageProcessorChain. This works fine, but for the path creating, this is what caused the issue for the path creation